src/main/java/org/ohdsi/authenticator/filter/JWTAuthenticationFilter.java

```diff
@@ -35,28 +37,50 @@ public JWTAuthenticationFilter(Authenticator authenticator) {
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
 
+        HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
         try {
-            HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
-            String authorize = httpRequest.getHeader(AuthorizationUtils.AUTHORIZE_HEADER);
-            if (StringUtils.isNotBlank(authorize) && authorize.startsWith(AuthorizationUtils.BEARER_AUTH)) {
-                String jwtToken = AuthorizationUtils.getToken(authorize);
+            String jwtToken = getToken(httpRequest);
+            if (StringUtils.isNotBlank(jwtToken)) {
                 String username = authenticator.resolveUsername(jwtToken);
                 if (Objects.nonNull(username) && SecurityContextHolder.getContext().getAuthentication() == null) {
-                    UserDetails userDetails = new User(username, "", getAuthorities(username));
+                    UserDetails userDetails = getUserDetails(username);
                     UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(userDetails, jwtToken,
                             userDetails.getAuthorities());
+                    onSuccessAuthentication(httpRequest, userDetails, authentication);
                     SecurityContextHolder.getContext().setAuthentication(authentication);
                 }
             }
             filterChain.doFilter(servletRequest, servletResponse);
         } catch (AuthenticationException e) {
-            logger.error("Authentication error:", e);
+            String method = httpRequest.getMethod();
+            if (!HttpMethod.OPTIONS.matches(method)) {
+                if (logger.isDebugEnabled()) {
```

Comment:
I would extract logging logic into a separate method. It would make code more clear.